llvm/lib/Target/AMDGPU/SIFoldOperands.cpp

```diff
@@ -1054,7 +1054,7 @@ void SIFoldOperands::foldOperand(
       // Don't fold if OpToFold doesn't hold an aligned register.
       const TargetRegisterClass *RC =
           TRI->getRegClassForReg(*MRI, OpToFold.getReg());
-      if (TRI->hasVectorRegisters(RC) && OpToFold.getSubReg()) {
+      if (RC && TRI->hasVectorRegisters(RC) && OpToFold.getSubReg()) {
```

Comment:
There's no context that can reach here where this can fail

Reply:
This would fail for unallocatable physical registers which shouldn't have been copy folding candidates to begin with

Reply:
Ok, then we can close this PR again

Reply:
Will adding an assert work?

Reply:
Hi Matt, yes an assert should also silence the issue:[1af318e](https://github.com/llvm/llvm-project/pull/83459/commits/1af318e8bacc13a5118303f6f444a8983e0ce9f0)